pkg/controllers/workloadrebalancer/workloadrebalancer_controller.go

```diff
@@ -35,32 +35,49 @@ import (
 
 	appsv1alpha1 "github.com/karmada-io/karmada/pkg/apis/apps/v1alpha1"
 	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util"
 	"github.com/karmada-io/karmada/pkg/util/names"
 )
 
 const (
 	// ControllerName is the controller name that will be used when reporting events.
 	ControllerName = "workload-rebalancer"
+
+	ttlAfterFinishedWorkerNum = 1
 )
 
 // RebalancerController is to handle a rebalance to workloads selected by WorkloadRebalancer object.
 type RebalancerController struct {
-	Client client.Client
+	Client             client.Client // used to operate WorkloadRebalancer resources from cache.
+	ControlPlaneClient client.Client // used to fetch arbitrary resources from api server.
```

Comment:
Here looks weird to have two control plane clients, one is used to fetch resources from the cache, and the other bypasses the cache.
Do we need to do that in such a way?

By the way, the comments are not precise enough, as the client(comes from controller-manager framework) not only operates resources against the cache.

Reply:
>  the comments are not precise enough, as the client(comes from controller-manager framework) not only operates resources against the cache.

got it, thanks.

> Here looks weird to have two control plane clients, ... Do we need to do that in such a way?

this is the same problem as next comment below.

Reply:
```suggestion
	Client             client.Client
	ClientWithOutCache versioned.Interface
```
How about using the raw client instead of the one from the controller-runtime framework?